articles/remapping.md

```diff
@@ -0,0 +1,77 @@
+### Remapping Names
+#### ROS 1
+In ROS 1 [Names](http://wiki.ros.org/Names) can be remapped to other names by passing [remapping arguments](http://wiki.ros.org/Remapping%20Arguments) via the command line. `roscpp` additionally allows remaps to be passed via Init as strings in STL containers. A ROS1 remap consists of two names: one that should be replaced with another.
+
+In ROS1 remapping works with FQN. When a name is to be checked for remappings, it is first expanded to a FQN. Both parts of the remapping rule are expanded to a FQN. Then a name is changed to the remap's second name only if it exactly matches the first name.
+
+#### ROS 2
+In ROS2 remapping will be more powerful. There are more types of remappings that will be supported
+
+- Namespace replacement
+
+ - Replace all or part of a namespace with another namespace
+ 
+- Basename replacement
+
+ - Replace all of the basename with another basename
+ 
+- Arbitrary substring replacement
+
+ - Replace any substring with another string
```

Comment:
It's not clear to me that we should support this in ROS 2. I mentioned it in the meeting for completeness. I'm not sure if it makes sense to support this in ROS 2, as I think that it might lead to really messy systems that rely on arcane remapping rules, i.e. I think the structure enforced by namespaces and basenames adds value over a free-for-all string replacement feature.